component/otelcol/consumer.go

```diff
@@ -12,6 +12,17 @@ type Consumer interface {
 	otelconsumer.Logs
 }
 
+// ConsumerArguments is a common Arguments type for Flow components which can
+// send data to otelcol consumers.
+//
+// It is expected to use ConsumerArguments as a block within the top-level
+// arguments block for a component.
+type ConsumerArguments struct {
+	Metrics []Consumer `river:"metrics,attr,optional"`
+	Logs    []Consumer `river:"logs,attr,optional"`
+	Traces  []Consumer `river:"traces,attr,optional"`
```

Comment:
We'll always use the `fanoutconsumer` package implementations for these internally, right?

Reply:
No, fanoutconsumer is only used when passing the list of consumers to the OpenTelemetry code. These will instead almost always be conusmers from the `lazyconsumer` package (unless one day there's different components which are otelcol-compatible that have their own approach for exposing consumers)